middleman-core/lib/middleman-core/core_extensions/i18n.rb

```diff
@@ -256,10 +273,20 @@ def parse_locale_extension(path)
     [locale, path, basename]
   end
 
-  LocalizedPageDescriptor = Struct.new(:path, :source_path, :locale) do
+  # Turn path into a page_id for the lookup map
+  # eg. "foo/bar/baz.foo.html" => "foo.bar.baz.foo"
```

Comment:
The idea is that it is then easy to use it as a key for lookup into I18n. So one could do: `{foo: {bar: {baz: {foo: "Text"}}}}` However thinking about it we could probably keep the slashes for `page_id` and do the conversion into I18n lookup key later in the process. But that's the idea behind.